src/docs/guides/laravel.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Deploy a Laravel App
+---
+
+Laravel is a PHP framework designed for web artisans who value simplicity and elegance in their code. It stands out for its clean and expressive syntax, and offers built-in tools to handle many common tasks found in modern web applications, making development smoother and more enjoyable.
+
+This guide covers how to deploy a Laravel app on Railway in four ways:
+
+1. [One-click deploy from a template](#one-click-deploy-from-a-template).
+2. [From a GitHub repository](#deploy-from-a-github-repo).
+3. [Using the CLI](#deploy-from-the-cli).
+4. [Deploy via Custom Scripts](#deploy-via-custom-scripts).
+
+## One-Click Deploy from a Template
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/fWEWWf)
+
+This template sets up a basic Laravel application along with a Postgres database on Railway. You can also choose from a <a href="https://railway.app/templates?q=laravel" target="_blank">variety of Laravel app templates</a> created by the community.
+
+We highly recommend that [you eject from the template after deployment](/guides/deploy#eject-from-template-repository) to create a copy of the repo on your GitHub account.
+
+**Note:** If you see a "500 Server Error" on your deployed URL, it’s likely because Laravel is missing the **APP_KEY** or other required environment variables. To fix this, go to the Railway dashboard and add the necessary environment variables—especially the **APP_KEY**—and then redeploy the application. You can generate a new **APP_KEY** using this [Laravel APP KEY generator](https://generate-random.org/laravel-key-generator).
+
+
+## Deploy from a GitHub Repo
+
+To deploy a Laravel app on Railway directly from GitHub, follow the steps below:
+
+1. Fork the basic <a href="https://github.com/railwayapp-templates/laravel" target="_blank">Laravel GitHub repo</a>. 
```

Comment:
What are your thoughts on having the guide start from a fresh Laravel app? Instead of forking an existing repo, it could start from new `composer create-project laravel/laravel my-project`?

What does the template have that is different from starting from scratch?

Reply:
It’s a combination of a few factors:

**Different Methods for Setting Up a Laravel App**:

- Users might be using tools like [Laravel Herd](https://herd.laravel.com/), which provides a UI for creating new projects. For instance, I use Herd a lot. 
- Alternatively, they could be using the [Laravel Installer](https://github.com/laravel/installer), where running laravel new <sample-app> in the terminal creates a new app.
For composer setups, I’ll need to add instructions to ensure the reader has composer installed.

**Starting from Scratch Can Be Overly Verbose**:

Building a fresh Laravel app from scratch makes the guide a lot more verbose which defeats the goal of reducing the time to deploy. For example, if using the "Deploy from GitHub" option (which I know is the most popular choice), I’d first need to guide the user through pushing their code to GitHub before proceeding with deployment instructions.

Instead, using a fork provides a quick and easy way to get a copy of the app in less than 30 seconds, saving time and making the user see it quickly on the Railway new project modal. This is why I also included an option to skip these steps if the user already has a GitHub repository.

Reply:
I have to agree with jr here, most people looking for a place to host their Laravel app would already have a project locally or on GitHub and wouldn't be interested in starting from scratch, so starting from the most common way to provision a Laravel app would give them a better understanding of what they need to do to their pre-existing local project so that deployment to Railway is a smooth process.

Reply:
I think the intended audience of this guide is someone who already knows Laravel and has a project that they want to host on Railway. We should assume some Laravel knowledge and explain how to specifically host on Railway with best practices.

Reply:
Like we all agree. This guide is for people that already know Laravel or have used it at least once and just want to host on Railway so it means they already have a running app locally or on GitHub. 

I just want to reiterate the following about the step by step approach in the guide:

- The GitHub repo option shows you how to deploy your Laravel app if it's on GitHub.
- The Deploy from CLI option shows you how to deploy your Laravel app if you have it locally like most people do. (if you don't want to push to GitHub).
- The Deploy via Custom scripts option uses both approach. It starts from locally and moves to having it on GitHub.

I'll just remove the fork option. I think that's the part that's making this debatable.

Reply:
I also changed the last section to **Set Up Database, Migrations, Crons and Workers** to logically take the reader from the initial steps to setting up all the other services for their Laravel apps to functional optimally.